scripts/build.sh

```diff
@@ -43,16 +43,24 @@ CERTS=""
 for ((ID=0; ID<FM_FED_SIZE; ID++));
 do
   mkdir $FM_CFG_DIR/server-$ID
-  base_port=$(echo "4000 + $ID * 10" | bc -l)
-  $FM_BIN_DIR/distributedgen create-cert --announce-address ws://localhost --out-dir $FM_CFG_DIR/server-$ID --base-port $base_port --name "Server-$ID" --password "pass$ID"
+  fed_port=$(echo "4000 + $ID * 10" | bc -l)
+  api_port=$(echo "4001 + $ID * 10" | bc -l)
+  if [ $ID -eq 0 ]; then
+    # Test that the ports will default to 4000 and 4001 if unspecified
+    $FM_BIN_DIR/distributedgen create-cert --p2p-url ws://localhost --api-url ws://localhost --out-dir $FM_CFG_DIR/server-$ID --name "Server-$ID" --password "pass$ID"
+  else
+    $FM_BIN_DIR/distributedgen create-cert --p2p-url ws://localhost:$fed_port --api-url ws://localhost:$api_port --out-dir $FM_CFG_DIR/server-$ID --name "Server-$ID" --password "pass$ID"
+  fi
   CERTS="$CERTS,$(cat $FM_CFG_DIR/server-$ID/tls-cert)"
 done
 CERTS=${CERTS:1}
 echo "Running DKG with certs: $CERTS"
 
 for ((ID=0; ID<FM_FED_SIZE; ID++));
 do
-  $FM_BIN_DIR/distributedgen run --out-dir $FM_CFG_DIR/server-$ID --certs $CERTS --password "pass$ID" &
+  fed_port=$(echo "4000 + $ID * 10" | bc -l)
```

Comment:
4000 (all round port numbers) is also often taken making this conflict for some people.